extensions/Lily/MoreControl.js

```diff
@@ -0,0 +1,1294 @@
+                menu: "targetsMyself",
+              },
+            },
+          },
+          "---",
+          {
+            opcode: "startBlocks",
+            blockType: Scratch.BlockType.CONDITIONAL,
+            text: "start blocks",
+            branchIconURI: this.junctionIcon,
+          },
+          {
+            opcode: "withoutScreenRefresh",
+            blockType: Scratch.BlockType.CONDITIONAL,
+            text: "without screen refresh",
+          },
+          {
+            opcode: "withScreenRefresh",
+            blockType: Scratch.BlockType.CONDITIONAL,
+            text: "with screen refresh",
+          },
+        ],
+        menus: {
+          types: {
+            acceptReporters: true,
+            items: ["frames", "seconds"],
+          },
+          lists: {
+            acceptReporters: true,
+            items: "_getLists",
+          },
+          targetsMyself: {
+            acceptReporters: true,
+            items: this._getTargets(true, true),
+          },
+          targets: {
+            acceptReporters: true,
+            items: this._getTargets(true, false),
+          },
+        },
+      };
+    }
+
+    /**
+     * Block Functions
+     */
+
+    switch(args, util) {
+      const blockId = this._getPeekStack(util.thread);
+      const blocks = util.thread.blockContainer;
+      const block = blocks.getBlock(blockId);
+      if (!block) return;
+
+      block.switchValue = args.SWITCH;
+      block.switchFalling = false;
+      block.switchResponse = false;
+      block.switchBroken = false;
+      block.runNextCase = false;
+
+      util.startBranch(1, false);
+    }
+
+    case(args, util) {
+      const blockId = this._getPeekStack(util.thread);
+      const outerC = this._getOuterCFromOpcode(
+        util.thread,
+        blockId,
+        "lmsSpAsMoreControl_switch"
+      );
+      if (!outerC) return;
+
+      if (
+        outerC.switchValue === args.CASE ||
+        outerC.switchFalling ||
+        outerC.runNextCase
+      ) {
+        if (outerC.switchBroken) return;
+        outerC.runNextCase = false;
+        outerC.switchFalling = true;
+        outerC.switchResponse = true;
+        util.startBranch(1, false);
+      }
+    }
+
+    default(args, util) {
+      const blockId = this._getPeekStack(util.thread);
+      const outerC = this._getOuterCFromOpcode(
+        util.thread,
+        blockId,
+        "lmsSpAsMoreControl_switch"
+      );
+      if (!outerC) return;
+
+      if (!outerC.switchResponse) {
+        if (outerC.switchBroken) return;
+        outerC.runNextCase = false;
+        outerC.switchFalling = true;
+        util.startBranch(1, false);
+      }
+    }
+
+    runNextCaseWhen(args, util) {
+      const blockId = this._getPeekStack(util.thread);
+      const outerC = this._getOuterCFromOpcode(
+        util.thread,
+        blockId,
+        "lmsSpAsMoreControl_switch"
+      );
+      if (!outerC) return;
+
+      if (args.CASE === outerC.switchValue) {
```

Comment:
This line (`659`) not allow integers and strings to match due to the **triple**(`===`) equals sign. I feel like a **double**(`==`) would be better.

Example: This does not work with `===`, because picking a random number returns an **integer** ~~(or float, I'm not entirely sure)~~ and the case inputs are **strings**.

This leads to a type mismatch which will always return false.

![block_11_29_2024-2_33_29 PM](https://github.com/user-attachments/assets/163de73c-4e14-43e3-9139-5ba71d2a0735)

However, changing this to `==` resolved the problem I was having.

This will always result in the **default** case being run, if added.

Side note: If this is an intended feature, ignore this; I know nothing about JavaScript. :p